src/ctapipe/irf/irfs.py

```diff
@@ -0,0 +1,357 @@
+    """
+    Creates a radially symmetric parameterization of the background rate in equidistant
+    bins of logarithmic reconstructed energy and field of view offset.
+    """
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+
+    def __call__(
+        self, events: QTable, obs_time: u.Quantity, extname: str = "BACKGROUND"
+    ) -> BinTableHDU:
+        background_rate = background_2d(
+            events=events,
+            reco_energy_bins=self.reco_energy_bins,
+            fov_offset_bins=self.fov_offset_bins,
+            t_obs=obs_time,
+        )
+        return create_background_2d_hdu(
+            background_2d=background_rate,
+            reco_energy_bins=self.reco_energy_bins,
+            fov_offset_bins=self.fov_offset_bins,
+            extname=extname,
+        )
+
+
+class PSF3DMaker(PSFMakerBase, DefaultFoVOffsetBins):
+    """
+    Creates a radially symmetric point spread function calculated in equidistant bins
+    of source offset, logarithmic true energy, and field of view offset.
+    """
+
+    source_offset_min = AstroQuantity(
+        help="Minimum value for Source offset",
+        default_value=u.Quantity(0, u.deg),
+        physical_type=u.physical.angle,
+    ).tag(config=True)
+
+    source_offset_max = AstroQuantity(
+        help="Maximum value for Source offset",
+        default_value=u.Quantity(1, u.deg),
+        physical_type=u.physical.angle,
+    ).tag(config=True)
+
+    source_offset_n_bins = Integer(
+        help="Number of bins for Source offset",
+        default_value=100,
+    ).tag(config=True)
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+        self.source_offset_bins = u.Quantity(
+            np.linspace(
+                self.source_offset_min.to_value(u.deg),
+                self.source_offset_max.to_value(u.deg),
+                self.source_offset_n_bins + 1,
+            ),
+            u.deg,
+        )
+
+    def __call__(
+        self, events: QTable, spatial_selection_applied: bool, extname: str = "PSF"
+    ) -> BinTableHDU:
+        psf = psf_table(
+            events=events,
+            true_energy_bins=self.true_energy_bins,
+            fov_offset_bins=self.fov_offset_bins,
+            source_offset_bins=self.source_offset_bins,
+        )
+        hdu = create_psf_table_hdu(
+            psf=psf,
+            true_energy_bins=self.true_energy_bins,
+            fov_offset_bins=self.fov_offset_bins,
+            source_offset_bins=self.source_offset_bins,
+            extname=extname,
+        )
+        # We also calculate a psf for IRFs including a spatial selection
+        # ("point-like IRF") to enable RAD_MAX cross-checks downstream.
+        # In that case, we have to change the header accordingly.
+        if spatial_selection_applied:
+            hdu.header["HDUCLAS3"] = "POINT-LIKE"
```

Comment:
point-like PSF does not make sense. Even if you include it in the output, which I still don't think is right, the PSF itself is not "point-like" if you didn't apply the theta cut to compute it, which you shouldn't.